web/ts/socket.ts

```diff
@@ -0,0 +1,141 @@
+const WS_INITIAL_RETRY_DELAY = 5000;
+const PAGE_LOADED = new Date();
+
+type MessageHandlerFn = (payload: object) => void;
+
+const RealtimeMessageTypes = {
+    RealtimeMessageTypeSubscribe: "subscribe",
+    RealtimeMessageTypeUnsubscribe: "unsubscribe",
+    RealtimeMessageTypeChannelMessage: "message",
+};
+
+export const realtime = {
+    _debugging: false,
```

Comment:
If you'd use TypeScript classes you could utilize TypeScripts `private` visibility keyword vor the `_<name>` attributes and methods.

Reply:
good idea! Refactoring to a singleton ts-class! 👍